### PR TITLE
Cleanup `schema.activate` logic

### PR DESCRIPTION
**Pull request by dimitri-yatsenko**
Fixed an issue when schemas were created with `create_schema=False`

### CODE REVIEW

File: datajoint/schemas.py [hunk 1]
```diff
@@ -273,43 +276,46 @@ def drop(self, force=False):
...
         """
         :return: true if the associated schema exists on the server
         """
-        self._assert_activation()
-        cur = self.connection.query("SHOW DATABASES LIKE '{database}'".format(database=self.database))
-        return cur.rowcount > 0
+        if self.database is None:
+            raise DataJointError("Schema must be activated first.")
+        return self.database is not None and (
+            self.connection.query("SHOW DATABASES LIKE '{database}'".format(
```
Comment (guzman-raphael) [inline datajoint/schemas.py:302]: We should avoid querying using syntax unique to MySQL.

```suggestion
            self.connection.query("SELECT schema_name FROM information_schema.schemata WHERE schema_name LIKE '{database}'".format(
```

**Review by guzman-raphael — CHANGES_REQUESTED** [review]
Looks good. Only have a minor update regarding MySQL-specific language.

File: datajoint/schemas.py (hunk 1 above)
Comment (dimitri-yatsenko) [reply datajoint/schemas.py]: perhaps here instead of `LIKE` we should just use `=`

Comment (guzman-raphael) [reply datajoint/schemas.py]: Yea wasn't sure if I missed something on first pass but looked like there were no wildcard operators used so should be good.

File: datajoint/schemas.py [hunk 2]
```diff
@@ -273,43 +276,46 @@ def drop(self, force=False):
...
         """
         :return: true if the associated schema exists on the server
         """
-        self._assert_activation()
-        cur = self.connection.query("SHOW DATABASES LIKE '{database}'".format(database=self.database))
-        return cur.rowcount > 0
+        if self.database is None:
+            raise DataJointError("Schema must be activated first.")
+        return self.database is not None and (
+            self.connection.query("SELECT schema_name FROM information_schema.schemata WHERE schema_name LIKE '{database}'".format(
```
Comment (guzman-raphael) [inline datajoint/schemas.py:302]:
```suggestion
            self.connection.query(
                "SELECT schema_name "
                "FROM information_schema.schemata "
                "WHERE schema_name = '{database}'".format(
```

**Review by guzman-raphael — CHANGES_REQUESTED** [review]
I've updated the suggestion. If you commit it we should be ready to merge.

**Review by guzman-raphael — APPROVED** [review]

**Pull request merged**